### PR TITLE
Build optional without QWebEngine

### PR DESCRIPTION
**Pull request by FlexW**
Basic auth is tested and works @er-vin ;)

This will be one step in the direction to allow building native clients for Apple M1 https://github.com/nextcloud/desktop/issues/2659

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,11 @@ include(ECMEnableSanitizers)
 
 set(synclib_NAME ${APPLICATION_EXECUTABLE}sync)
 
-find_package(Qt5 5.12 COMPONENTS Core Network Xml Concurrent WebEngineWidgets WebEngine REQUIRED)
+find_package(Qt5 5.12 COMPONENTS Core Network Xml Concurrent REQUIRED)
+if(WITH_WEBENGINE)
+  find_package(Qt5 5.12 COMPONENTS WebEngineWidgets WebEngine REQUIRED)
```
Comment (mgallien) [inline src/CMakeLists.txt:11]: I would have just made them OPTIONAL dependencies and then build can use the found variable (`Qt5WebEngine_FOUND`)
I get that we have both WebEngine and WebEngineWidgets module but adding one more option feels strange for me if teh goal is to downgrade the dependency requirement from mandatory to optional

Comment (FlexW) [reply src/CMakeLists.txt]: Makes sense. Done.

**Review by mgallien — APPROVED** [review]
cool, thanks !

**Comment by mgallien** [comment]
> cool, thanks !

Please do not forget to clean history before the merge

**Review by allexzander — APPROVED** [review]

**Pull request merged**